src/wallet/wallet.cpp

```diff
@@ -3950,6 +3951,65 @@ std::vector<std::string> CWallet::GetDestValues(const std::string& prefix) const
     return values;
 }
 
+bool CWallet::Verify(std::string wallet_file, bool salvage_wallet)
+{
+    // Do some checking on wallet path. It should be either a:
+    //
+    // 1. Path where a directory can be created.
+    // 2. Path to an existing directory.
+    // 3. Path to a symlink to a directory.
+    // 4. For backwards compatibility, the name of a data file in -walletdir.
+    LOCK(cs_wallets);
+    fs::path wallet_path = fs::absolute(wallet_file, GetWalletDir());
+    fs::file_type path_type = fs::symlink_status(wallet_path).type();
+    if (!(path_type == fs::file_not_found || path_type == fs::directory_file ||
+          (path_type == fs::symlink_file && fs::is_directory(wallet_path)) ||
+          (path_type == fs::regular_file && fs::path(wallet_file).filename() == wallet_file))) {
+        return InitError(strprintf(
+            _("Invalid -wallet path '%s'. -wallet path should point to a directory where wallet.dat and "
+              "database/log.?????????? files can be stored, a location where such a directory could be created, "
+              "or (for backwards compatibility) the name of an existing data file in -walletdir (%s)"),
+            wallet_file, GetWalletDir()));
+    }
+
+    // Make sure that the wallet path doesn't clash with an existing wallet path
+    std::set<fs::path> wallet_paths;
```

Comment:
commit: [wallet] Add CWallet::Verify function

Using a set here is unnecessary, can just do a direct equality check.

Reply:
Yes, much better idea.